### PR TITLE
[JH][COM-99] Mass Aggregate Finalization

### PR DESCRIPTION
**Pull request by jharris99-git**
PR for the Mass Aggregate physics system.

### CODE REVIEW

File: Physics/include/pointmass.h
Comment (le-rich) [inline Physics/include/pointmass.h:1]: Your tab size is larger compared to the rest of the repo, 1 tab length or 4 spaces where the rest of the repo is 2 space tabs.

File: Physics/src/pointmass.cpp
Comment (le-rich) [inline Physics/src/pointmass.cpp:51]: Nitpick: Wrap the cast in brackets like above

File: Physics/src/pointmass.cpp
Comment (le-rich) [inline Physics/src/pointmass.cpp:65]: function is named getPosition but it's setting.

File: Physics/src/pointmass.cpp
Comment (le-rich) [inline Physics/src/pointmass.cpp:67]: If the purpose of this function is to assign some other object, we should rename the function

Comment (le-rich) [reply Physics/src/pointmass.cpp:67]: Typically if you want to make it very clear that the pointer is an "out" variable we prefix the variable name with OUT or out 

i.e. outPosition

File: Physics/src/pointmass.cpp
Comment (le-rich) [inline Physics/src/pointmass.cpp:89]: ditto points above

File: Physics/src/pointmass.cpp
Comment (le-rich) [inline Physics/src/pointmass.cpp:113]: ditto points above

**Review by le-rich — CHANGES_REQUESTED** [review]
Tab spacing nitpicks, some comments about "get"

File: Physics/include/pointmass.h
Comment (jharris99-git) [reply Physics/include/pointmass.h:1]: Looking around the repo, renderer has files on the repo that have _non-tab_ 4 space lines. Can you show pictures on your end comparing? Physics project files are using tabs for our tab sizing, not spaces, as the tabs can't be subdivided when selecting,

File: Physics/src/pointmass.cpp
Comment (jharris99-git) [reply Physics/src/pointmass.cpp]: Calling other physics contributors to weigh in on naming of the void getters @imnorookie @jchouhan19

File: Physics/src/pointmass.cpp
Comment (jharris99-git) [reply Physics/src/pointmass.cpp]: Resolved

File: Physics/src/pointmass.cpp
Comment (le-rich) [inline Physics/src/pointmass.cpp:36]: Wrap Left hand side cast in brackets like on line 49

File: Physics/src/pointmass.cpp
Comment (le-rich) [inline Physics/src/pointmass.cpp:12]: wrap right hand side cast like other examples

**Review by le-rich — CHANGES_REQUESTED** [review]
Brackets on new real casts

File: Physics/src/pointmass.cpp
Comment (le-rich) [inline Physics/src/pointmass.cpp:36]: left hands side wrap (((real) 1.0) / inverseMass)

**Review by le-rich — CHANGES_REQUESTED** [review]

**Review by le-rich — DISMISSED** [review]
Approved but lets not keep creeping in new code to PRs

File: Physics/src/pointmass.cpp
Comment (imnorookie) [reply Physics/src/pointmass.cpp]: I think keeping it as a getter is fine but renaming the formal parameter to "out" for clarity reasons is a good idea thaWanks foWor that idea RichaWrd :333. 

This accessor is like this for performance reasons, so it isn't returning a new Vector3 object every time we're checking and applying positional updates. This is because return happens by value in CPP, meaning it can be very costly if we are simulating a ton of physics bodies in a single PhysicsWorld time-step. We'd be creating a new Vector3 object every single time.

My compromise would be to rename the formal parameter to "out" and then also make a second getter that returns a Vector3 copy as well, should someone interfacing with our Physics code want it. BeWest of BoWoth woWorlds! Fix it nao!!!!

**Review by imnorookie — APPROVED** [review]
This is good, thanks for all of your hard worksies ~~ *!!*

Make the change to the position getter if you have time and reping me for another approval or just approve it now if you can't be bothered for Great Good *!!*

**Comment by jharris99-git** [comment]
> Make the change to the position getter if you have time
Already done immediately after the suggestion, since it was a good suggestion. PR is ready for merge, waiting on approvals

**Comment by imnorookie** [comment]
> > Make the change to the position getter if you have time
> 
> Already done immediately after the suggestion, since it was a good suggestion. PR is ready for merge, waiting on approvals

okay dokie ! ^.^

**Review by le-rich — APPROVED** [review]
Best practice in C++ to use prefix increment:

++i 

Approved but let's add this in the future

**Pull request merged**
